class.jetpack-onboarding-end-points.php

```diff
@@ -564,44 +555,61 @@ static function insert_widget_in_sidebar( $widget_id, $widget_data, $sidebar ) {
 		$sidebars_widgets[ $sidebar ][] = $widget_id . '-' . $next_key;
 
 		// Add the new widget instance
-		$widget_instances[ $next_key ] = $widget_data;
+		$widget_instances[ $next_key ] = $widget_options;
 
 		// Store updated sidebars, widgets and their instances
 		update_option( 'sidebars_widgets', $sidebars_widgets );
 		update_option( 'widget_' . $widget_id, $widget_instances );
 	}
 
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		$excluded_keys = array(
+			'wp_inactive_widgets',
+			'array_version',
+		);
+
+		foreach ( $excluded_keys as $key ) {
+			if ( isset( $active_sidebars[ $key ] ) ) {
+				unset( $active_sidebars[ $key ] );
+			}
+		}
+
+		if ( empty( $active_sidebars ) ) {
+			return false;
+		}
+
+		return array_shift( array_keys( $active_sidebars ) );
+	}
+
 	static function add_business_address() {
 		check_ajax_referer( self::AJAX_NONCE, 'nonce' );
-		$active_sidebars = get_option( 'sidebars_widgets' );
-		$widget_options = get_option( 'widget_widget_contact_info' );
-		if( isset( $active_sidebars['sidebar-1'] ) &&
-			( ! isset( $widget_options[1] ) || str_replace(array("\n", "\r"), '', $widget_options[1]['address'] )== '3999 Mission Boulevard, San Diego CA 92109' ) // default address
-			) {
-			$title =  wp_unslash( $_REQUEST['business_name'] );
-			$address = wp_unslash( $_REQUEST['business_address_1'] . ' ' .
-				$_REQUEST['business_address_2']. ' ' .
-				$_REQUEST['business_city']. ' ' .
-				$_REQUEST['business_state']. ' ' .
-				$_REQUEST['business_zip']. ' ' );
-			$phone = wp_unslash( $_REQUEST['phone'] );
-			$hours = wp_unslash( $_REQUEST['hours'] );
-			$widget_options[1] = array( 'title' => $title, 'address' => $address, 'phone' => $phone, 'hours' => $hours, 'showmap' => false );
-			$widget_options[0] = 1;
-			update_option( 'widget_widget_contact_info', $widget_options ); //update widget default options
-			if ( ! self::have_contact_info_widget( $active_sidebars['sidebar-1'] ) ) {
-
-				array_push( $active_sidebars['sidebar-1'], 'widget_contact_info-1' );
-				update_option( 'sidebars_widgets', $active_sidebars ); //update sidebars
-				wp_send_json_success( array( 'widget' => true, 'updated' => true ) );
-				die();
-			}
 
-			wp_send_json_success( array( 'widget' => false, 'updated' => true ) );
+		$first_sidebar = self::get_first_sidebar();
+		if ( $first_sidebar ) {
+			$title = wp_unslash( $_REQUEST['business_name'] );
+			$address = wp_unslash(
+				$_REQUEST['business_address_1'] . ' ' .
+				$_REQUEST['business_address_2'] . ' ' .
+				$_REQUEST['business_city'] . ' ' .
+				$_REQUEST['business_state'] . ' ' .
+				$_REQUEST['business_zip']
+			);
+			$widget_options = array(
+				'title'   => $title,
+				'address' => $address,
+				'phone'   => '',
+				'hours'   => '',
+				'showmap' => false
+			);
+
+			self::insert_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
```

Comment:
yeah, that's also a good point. But right now it's not so easy to do it (skip the step) because we don't have that info passed from the php backend to the react frontend, who determines when to skip a step or not, and today it's the 'delivery' day so ... why don't add the existing-widget check again, and then, if @jessefriedman likes the idea, add all the logic to skip the step afterwards?